### ISSUE TITLE
additional token auth

### ISSUE THREAD

**Issue by Exordian** [coduno/platform#9]

**Comment by victorbalan** [comment]
[this](https://github.com/coduno/platform/blob/master/platform-server-kernel/src/main/java/uno/cod/platform/server/core/service/InvitationService.java#L119) is for invitation only. So we need a normal login like this.

**Comment by flowlo** [comment]
Yes, the current solution (invitation only) needs to be generalized, so one can use tokens instead of a password everywhere™.

**Comment by Exordian** [comment]
whats the usecase for that token?

**Comment by flowlo** [comment]
As a means to authenticate when using a CLI.

**Comment by Exordian** [comment]
we can generate some static api token for every user, which can be obtained from User => Settings on the platform, but i'd delay that feature until we build some sort of CLI

**Comment by flowlo** [comment]
Yes, the scenario about managing one's token via User Settings fits.

However, we're using tokens already with invitations: An invitation token is an authentication token for the user. The user just has not generated it explicitly, but we do, once we send the invitation.

We might not need the "User Settings" UI just now, but we should be aware that tokens might be used for other stuff than just inviting someone.

**Comment by flowlo** [comment]
Ideally the token we're talking about here should be valid...

 * for Basic Authentication (for example to `curl` our API).
 * from inside a Cookie (our SPA uses cookies because they cannot be tampered using JS).
 * for Bearer Authentication (for a CLI).
 * ...

They should expire after some time, and maybe be signed by us. [JWTs](https://jwt.io/) look like they could do all that. WDYT?
